Skip attestation verification

dailinsubjam opened the pull request:
Closes https://app.asana.com/1/1208976916964769/project/1209393353274209/task/1211858367366881?focus=true
<!-- These comments should help create a useful PR message, please delete any remaining comments before opening the PR. -->
<!-- If there is no issue number make sure to describe clearly *why* this PR is necessary. -->
<!-- Mention open questions, remaining TODOs, if any -->

### This PR:
- Based on https://github.com/EspressoSystems/espresso-tee-contracts/pull/26 
- Updated `BatchAuthenticator.sol` to expose the new function
- Modified batcher logic to:
    - Extract PCR0 hash from attestation document and Calculate enclave address from public key as the input for the new function
    - Call new gas-optimized function
- Reduce [L1 gas limit from 45M to 16M](https://github.com/EspressoSystems/optimism-espresso-integration/pull/263/commits/b64b2691a3123ecb66e3d5f7f9eb416869892229)
<!-- Describe what this PR adds to this repo and why -->
<!-- E.g. -->
<!-- * Implements feature 1 -->
<!-- * Fixes bug 3 -->

### This PR does not:
- Create off-chain verification application / have caff-node stop when verification is not valid yet, these will be in separate tasks
<!-- Describe what is out of scope for this PR, if applicable. Leave this section blank if it's not applicable -->
<!-- This section helps avoid the reviewer having to needlessly point out missing parts -->
<!-- * Implement feature 3 because that feature is blocked by Issue 4 -->
<!-- * Implement xyz because that is tracked in issue #123. -->
<!-- * Address xzy for which I opened issue #456 -->

### Key places to review:
<!-- Describe key places for reviewers to pay close attention to -->
<!-- * file.rs, `add_integers` function -->
<!-- Or directly comment on those files/lines to make it easier for the reviewers -->

### How to test this PR:
CI should be able to cover the test
<!-- Optional, uncomment the above line if this is relevant to your PR -->
<!-- If your PR is fully tested through CI there is no need to add this section -->
<!-- * E.g. `just test` -->

<!-- ### Things tested -->
<!-- Anything that was manually tested (that is not tested in CI). -->
<!-- E.g. building/running of docker containers. Changes to docker demo, ... -->
<!-- Especially mention anything untested, with reasoning and link an issue to resolve this. -->

<!-- Complete the following items before creating this PR -->
<!-- [ ] Issue linked or PR description mentions why this change is necessary. -->
<!-- [ ] PR description is clear enough for reviewers. -->
<!-- [ ] Documentation for changes (additions) has been updated (added).  -->
<!-- [ ] If this is a draft it is marked as "draft".  -->
<!-- [ ] For integration projects, make sure it won't break backward compatibility. -->

<!-- Details on maintaining backward compatibility for integration projects: -->
<!-- * Try to keep changes additive: add new, optional methods, flags, or parameters instead of modifying or removing existing functionality. -->
<!-- * If modification is necessary, it should be either a clear bug fix or guarded by a config/feature flag.  -->
<!-- * Follow Open Closed Principle and Interface Segregation Principle, clients should not be forced to depend on interfaces they do not use. -->

<!-- To make changes to this template edit https://github.com/EspressoSystems/.github/blob/main/PULL_REQUEST_TEMPLATE.md -->

dailinsubjam committed:
Update espresso-tee-contracts submodule to sishan/skip-attestation-verification

dailinsubjam committed:
Skip attestation verification to reduce gas costs

dailinsubjam committed:
Reduce L1 gas limit from 45M to 16M

Commit:
Update snapshots for registerSignerWithoutAttestationVerification

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/EspressoSystems/optimism-espresso-integration?pullRequest=263) <br/>All committers have signed the CLA.

dailinsubjam committed:
Ignore lib/automate submodule directory

dailinsubjam committed:
fix CI

dailinsubjam committed:
Update espresso-tee-contracts submodule

Remove onlyOwner modifier from registerSignerWithoutAttestationVerification

QuentinI commented on espresso/docker/l1-geth/devnet-genesis-template.json:
```diff
@@ -36,7 +36,7 @@
   "nonce": "0x0",
   "timestamp": "0x685c6a58",
   "extraData": "0x",
-  "gasLimit": "0x2aea540",
```
What is the reason for this change?

dailinsubjam replied:
I assume it does not need that much gas so I want to test it with lower gas limit, but let me switch it back before I really run the test.

QuentinI reviewed (approved):
LGTM

dailinsubjam committed:
keep large gasLimit

dailinsubjam committed:
circleci: Enable workflow on all branches via API trigger

Allow CircleCI main workflow to run on any branch when triggered via API,
not just webhook triggers. This enables go-lint and go-tests to run on
feature branches.

dailinsubjam committed:
Regenerate semver-lock.json after rebase

The initCodeHash for BatchAuthenticator needed to be regenerated
after rebasing onto celo-integration-rebase-14.1.
